dotfiles/install.sh

```diff
@@ -0,0 +1,138 @@
+#!/bin/bash
+# Install dotfiles for gptme agent
+# Creates symlinks for global git hooks and configuration
+#
+# Safety: This script includes checks to prevent accidentally running
+# on non-agent systems and overwriting user configurations.
+
+set -e
+
+DOTFILES_DIR="$(cd "$(dirname "$0")" && pwd)"
+
+# Colors for output
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
+NC='\033[0m' # No Color
+
+# --- Safety Check: Detect if running on agent system ---
+# Override with DOTFILES_FORCE=1 to bypass safety check
+check_agent_environment() {
+    # If forced, skip check
+    if [ "$DOTFILES_FORCE" = "1" ]; then
+        echo -e "${YELLOW}⚠️  Safety check bypassed (DOTFILES_FORCE=1)${NC}"
+        return 0
+    fi
+
+    # Check common agent indicators
+    local is_agent=false
+
+    # Check 1: Agent-specific directories
+    if [ -d "$HOME/bob" ] || [ -d "$HOME/alice" ]; then
+        is_agent=true
+    fi
+
+    # Check 2: gptme.toml with agent configuration
+    if [ -f "$DOTFILES_DIR/../gptme.toml" ]; then
+        if grep -q '\[agent\]' "$DOTFILES_DIR/../gptme.toml" 2>/dev/null; then
+            is_agent=true
+        fi
+    fi
+
+    # Check 3: Running in VM (common for agents)
+    if [ -f "/sys/class/dmi/id/chassis_type" ]; then
+        chassis_type=$(cat /sys/class/dmi/id/chassis_type 2>/dev/null || echo "")
+        # Type 1 = Other/VM
+        if [ "$chassis_type" = "1" ]; then
+            is_agent=true
+        fi
+    fi
+
+    # Check 4: Explicit agent username patterns
+    if [[ "$USER" =~ ^(bob|alice|agent)$ ]]; then
+        is_agent=true
+    fi
```

Comment:
This is not a reliable check. We cannot rely on rare names like alice and bob, chassis/VM, etc. 
What we could check is that the $USER is equal to the agent name, or contains the agent name, or vice versa, or something, but that might not be sufficient. 
We really need to think better about this, maybe check for the presence of a `*-autonomous.service`? but that's also brittle (and would come from dotfiles).
We should probably set a global env var somewhere indicating an agent environment, but that's the kind of thing we would do in dotfiles... Tricky.